### PR TITLE
implement '$not' regex

### PR DESCRIPTION
**Pull request by tsetsoo**
Implementing `$not` filter operator with regex. It has the opposite function of `$regex`.

Part of #273

### CODE REVIEW

File: README.md
Comment (smyrman) [inline README.md:1001]: We should stick to the MongoDB definition if implementing this:

https://docs.mongodb.com/manual/reference/operator/query/not/

File: README.md
Comment (smyrman) [inline README.md:1046]: To get the exact functionality you are after, you would need to do:
`{a: {$not: {$regex: "fo[o]{1}"}}}`

**Review by smyrman — CHANGES_REQUESTED** [review]

File: README.md
Comment (smyrman) [reply README.md:1046]: Ah.. I now see that `$not: string` is allowed as a short-hand for `$not: {$regex: string}` . I suppose that's fair enough.

File: README.md
Comment (smyrman) [reply README.md:1001]: Did not see your comment in https://github.com/rs/rest-layer/issues/273#issuecomment-925864211 earlier. Maybe extend the text somewhat:

```suggestion
The operator `$not` functions as an opposite operator to `$regex`. Unlike MongoDB, we do not allow `$not` as a general negation operator.
```

File: schema/query/predicate_parser.go
Comment (smyrman) [inline schema/query/predicate_parser.go:266]: Maybe we could instead return `$Regex{Field: field, Value: re, Negated: true}`.

This would give a clear path-way for how to support the `$not` operator in general later, if there is a need for that.

Comment (tsetsoo) [reply schema/query/predicate_parser.go]: great suggestion, thanks

File: README.md
Comment (tsetsoo) [reply README.md:1046]: `$not: {$regex: string}` was added in Mongo 4.0.6 so I suppose `$not: string` is the conventional way. Regardless, if we decide to add support for `$not` as a general negation operator, both will be supported.

File: README.md
Comment (tsetsoo) [reply README.md]: done

**Comment by tsetsoo** [comment]
@smyrman Do you think we can go ahead and merge this PR? The regex functionality is needed regardless of whether we decide to support the ability to _not_ the whole expression.

**Comment by smyrman** [comment]
Let's go ahead. Could you please squash your changes into one commit?

Example:
```
git fetch
git rebase -i origin/master
```
mark second commit as "fixup".

**Comment by tsetsoo** [comment]
Done

**Review by smyrman — DISMISSED** [review]

File: schema/query/predicate_parser.go
Comment (smyrman) [inline schema/query/predicate_parser.go:253]: can be written without a branch condition
```suggestion
			negated := label == opNot 
```

Comment (smyrman) [reply schema/query/predicate_parser.go:253]: maybe fix this as well (squash it in, then I can merge).

Comment (tsetsoo) [reply schema/query/predicate_parser.go]: done

**Review by smyrman — APPROVED** [review]

**Comment by smyrman** [comment]
@rs: Seams travis is no longer working, so I can't merge this:

image

**Comment by rs** [comment]
I removed the requirement for travis check before merging waiting for a github actions migration.

**Pull request merged**
